### ISSUE TITLE
Hard system lock when using the alt-fire for the weapon in slot 7 while in 3rd person mode.

### ISSUE THREAD

**Issue by skdursh** [jekyllgrim/Alice-Toys-of-Madness#7]
Windows 10
Using GZDoom 4.13.2
RTX 4070 Super - i7-8700k - 64GB RAM 3333MHz
Playing the WAD "What Lies Beneath"

Gave the mod a try last night. Was thoroughly enjoying it until I got the slot 7 weapon (sorry I don't know the name of it). Upon trying the alt fire while in 3rd person mode, the attack where a big purple circle appears on the ground and then you have to "fill" it up to launch the attack, after charging the attack and releasing I could see an attack animation beginning where it looked like projectiles were going to rain down from the sky, but as soon as the projectiles appeared the game crashed and caused a complete and total system lock. I had to hard reset the computer to resume using it in any capacity. Not sure if there is a crash log available anywhere for the game to give you a better idea of what happened, but I at least wanted to mention that this happened to me.

**Comment by jekyllgrim** [comment]
So, there are a few things to unpack here.

First of all, I'm not observing any issues with the eyestaff on my end. Using it in third person, including in What Lies Beneath levels, works fine for me. So, I would first need to know whether you can reproduce this issue consistently. If you can, there are probably some kind of specific conditions to it. Perhaps, it has to be used in a specific place of a specific level, or something like that. If you can figure out exact conditions, describe them to me, because simply using the weapon doesn't trigger any issues for me.

Second, regarding the crash. Normally, GZDoom errors can only have two outcomes: a VM abort (game closes to console) or a Very Fatal Error (game closes to an error window with a memory address). The former is always a scripting issue; the latter can be a result of a scripting issue, but ultimately it's a GZDoom issue, because it means that GZDoom couldn't properly handle the problem. Ideally GZDoom must never crash and in case of any error should close to the console.

However, GZDoom itself cannot cause a system lock-up, neither can a mod do that. That means that your system for some reason couldn't handle what was likely a GZDoom fatal error. It could be a driver problem, a hardware problem, something resulting from overclocking, or something else similar.

So, to sum up:
1. If there's an actual scripting problem behind this, I will need to know what exactly it is. For that I need to know the exact conditions for triggering the bug, if you can figure them out. Add the text `logfile log.txt` in the autoexec.cfg file in your GZDoom folder, this will output the console log in a text file (If you don't have autoexec.cfg, simply create it, it should be a text file with the .cfg extension); this will help with debugging if you manage to trigger it.
2. If this scripting problem causes a GZDoom hard crash, this will also need to be turned into a bug report for GZDoom.
3. If this issue freezes your system, that's completely beyond my or GZDoom team's reach.

**Comment by skdursh** [comment]
I will mess around with it again tomorrow night when I have some more free time to see if I can reproduce the issue and will hopefully have something more concrete to share in regards to the crash if I encounter it again. I will say though that as far as potential issues go, I'm not overly convinced that it's a problem with my system itself. My hardware is resonably beefy by most modern standards, the drivers I am currently using are generally considered by the Nvidia community at large to be the most stable recent drivers available and while I do have a slight overclock on the CPU, I've had the same profile enabled for well over a year now and it has been exceptionally stable in every game and application I've tested it on to date, including GZDoom. In any case, I'll share what I can sometime over the next day or two.
